### PR TITLE
 BAEL-1371 - Guide to Cipher class 

### PR DESCRIPTION
**Pull request by IkoroVictor**

### CODE REVIEW

File: spring-core/src/main/java/com/baeldung/beaninjectiontypes/config/AppConfig.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package com.baeldung.beaninjectiontypes.config;
```
Comment (KevinGilmore) [inline spring-core/src/main/java/com/baeldung/beaninjectiontypes/config/AppConfig.java]: These spring-core files should not be part of this PR.

File: core-java/src/test/java/com/baeldung/cipher/EncryptorTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+import java.io.ByteArrayInputStream;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+
+public class EncryptorTest {
```
Comment (KevinGilmore) [inline core-java/src/test/java/com/baeldung/cipher/EncryptorTest.java]: Rename class to EncryptorUnitTest

File: core-java/src/test/java/com/baeldung/cipher/EncryptorTest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                "kXCqKWpnZ/Oc75EUcMi6HztaW8abUMlYXPIgkV2F7YanHOB7K4f7OOLjiz8DTPFf\n" +
+                "jC9UeuErhaA/zzWi8ewMTFZW/WshOrm3fNvcMrMLKtH534JKvcdMg6qIdjTFINIr\n" +
+                "evnAhf0cwULaebn+lMs8Pdl7y37+sfluVok=\n" +
+                "-----END CERTIFICATE-----";
+    }
+
+    @Test
+    public void givenEncryptionKey_thenMessageCanBeEncrypted() throws Exception {
+        byte[] encryptedMessage = encryptor.encryptMessage(message.getBytes(),encKeyString.getBytes());
+        Assert.assertNotNull(encryptedMessage);
```
Comment (KevinGilmore) [inline core-java/src/test/java/com/baeldung/cipher/EncryptorTest.java]: We're standardizing on AssertJ assertions (rather than JUnit assertions).

File: core-java/src/test/java/com/baeldung/cipher/EncryptorTest.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                "MwBlc8ksFJHB0rtf9pmf8Oza9E0Bynlq/4/Kb1x+d+AyhL7oK9tQwB24uHOueHi1\n" +
+                "C/iVv8CSWKiYe6hzN1txYe8rAgMBAAEwDQYJKoZIhvcNAQEFBQADgYEAASPdjigJ\n" +
+                "kXCqKWpnZ/Oc75EUcMi6HztaW8abUMlYXPIgkV2F7YanHOB7K4f7OOLjiz8DTPFf\n" +
+                "jC9UeuErhaA/zzWi8ewMTFZW/WshOrm3fNvcMrMLKtH534JKvcdMg6qIdjTFINIr\n" +
+                "evnAhf0cwULaebn+lMs8Pdl7y37+sfluVok=\n" +
+                "-----END CERTIFICATE-----";
+    }
+
+    @Test
+    public void givenEncryptionKey_thenMessageCanBeEncrypted() throws Exception {
```
Comment (KevinGilmore) [inline core-java/src/test/java/com/baeldung/cipher/EncryptorTest.java]: Test method names should use the format givenX_whenY_thenZ (the "givenX" is optional).
This one and others are missing the "whenY".

File: core-java/src/test/java/com/baeldung/cipher/EncryptorTest.java (hunk 3 above)
Comment (KevinGilmore) [inline core-java/src/test/java/com/baeldung/cipher/EncryptorTest.java]: Insert a blank line between the "when" and "then" portions of your tests. In other words, between the code being tested and the beginning of assertions.

**Comment by IkoroVictor** [comment]
Hi @KevinGilmore, thanks for the feedback. I have updated the pull request accordingly.

File: core-java/src/test/java/com/baeldung/cipher/EncryptorUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
+package com.baeldung.cipher;
+
+
+import com.baeldung.cipher.Encryptor;
```
Comment (KevinGilmore) [inline core-java/src/test/java/com/baeldung/cipher/EncryptorUnitTest.java]: We have a few unused imports here -- let's clean those up.

File: core-java/src/test/java/com/baeldung/cipher/EncryptorUnitTest.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+                "MwBlc8ksFJHB0rtf9pmf8Oza9E0Bynlq/4/Kb1x+d+AyhL7oK9tQwB24uHOueHi1\n" +
+                "C/iVv8CSWKiYe6hzN1txYe8rAgMBAAEwDQYJKoZIhvcNAQEFBQADgYEAASPdjigJ\n" +
+                "kXCqKWpnZ/Oc75EUcMi6HztaW8abUMlYXPIgkV2F7YanHOB7K4f7OOLjiz8DTPFf\n" +
+                "jC9UeuErhaA/zzWi8ewMTFZW/WshOrm3fNvcMrMLKtH534JKvcdMg6qIdjTFINIr\n" +
+                "evnAhf0cwULaebn+lMs8Pdl7y37+sfluVok=\n" +
+                "-----END CERTIFICATE-----";
+    }
+
+    @Test
+    public void givenEncryptionKey_whenMessageIsPassedToEncryptor_thenMessageIseEncrypted() throws Exception {
```
Comment (KevinGilmore) [inline core-java/src/test/java/com/baeldung/cipher/EncryptorUnitTest.java]: Typo in last part of method name (IseEncrypted should be IsEncrypted)

File: core-java/src/test/java/com/baeldung/cipher/EncryptorUnitTest.java [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+    @Test
+    public void givenCertificateWithPublicKey_whenMessageIsPassedToEncryptor_thenMessageIsEncrypted() throws Exception {
+        CertificateFactory factory = CertificateFactory.getInstance("X.509");
+        InputStream is = new ByteArrayInputStream(certificateString.getBytes());
+        X509Certificate certificate = (X509Certificate) factory.generateCertificate(is);
+
+        byte[] encryptedMessage = encryptor.encryptMessage(message.getBytes(),certificate);
+        
+        Assert.assertNotNull(encryptedMessage);
```
Comment (KevinGilmore) [inline core-java/src/test/java/com/baeldung/cipher/EncryptorUnitTest.java]: We need to be using the AssertJ library to do assertions, as per recent site policy. The library should already be included in the pom and ready to use.
